SPECS/dpdk/dpdk.spec

```diff
@@ -0,0 +1,523 @@
+%ifarch x86_64
+%define machine_arch x86_64
+%define machine_tmpl native
+%define machine default
+%endif
+%ifarch i686
+%define machine_arch i686
+%define machine_tmpl native
+%define machine default
+%endif
+%ifarch aarch64
+%define machine_arch arm64
+%define machine_tmpl armv8a
+%define machine armv8a
+%endif
+%ifarch ppc64le
+%define machine_arch ppc_64
+%define machine_tmpl power8
+%define machine power8
+%endif
+%bcond_without examples
+# Add option to build without tools
+%bcond_without tools
+Summary:        Set of libraries and drivers for fast packet processing
+Name:           dpdk
+Version:        18.11.2
+Release:        5%{?dist}
+Epoch:          2
+License:        BSD AND LGPLv2 AND GPLv2
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+Group:          System Environment/Libraries
+URL:            https://dpdk.org
+Source0:        https://fast.%{name}.org/rel/%{name}-%{version}.tar.xz
+Patch0:         app-pie.patch
+Patch1:         fcf-protection.patch
+Patch2:         dpdk-rte-ether-align.patch
+BuildRequires:  gcc
+BuildRequires:  kernel-headers
+BuildRequires:  libnuma-devel
+BuildRequires:  libpcap-devel
+BuildRequires:  mariner-rpm-macros
+BuildRequires:  python3-sphinx
+BuildRequires:  zlib-devel
+#
+# The DPDK is designed to optimize througput of network traffic using, among
+# other techniques, carefully crafted assembly instructions.  As such it
+# needs extensive work to port it to other architectures.
+#
+ExclusiveArch:  x86_64 i686 aarch64 ppc64le
+
+%description
+The Data Plane Development Kit is a set of libraries and drivers for
+fast packet processing in the user space.
+
+%package devel
+Summary:        Data Plane Development Kit development files
+Requires:       %{name}%{?_isa} = %{epoch}:%{version}-%{release}
+Requires:       python3
+
+%description devel
+This package contains the headers and other files needed for developing
+applications with the Data Plane Development Kit.
+
+%package doc
+Summary:        Data Plane Development Kit API documentation
+BuildArch:      noarch
+
+%description doc
+API programming documentation for the Data Plane Development Kit.
+
+%if %{with tools}
+%package tools
+Summary:        Tools for setting up Data Plane Development Kit environment
+Requires:       %{name} = %{epoch}:%{version}-%{release}
+Requires:       findutils
+Requires:       iproute
+Requires:       kmod
+Requires:       pciutils
+Requires:       python3-pyelftools
+
+%description tools
+%{summary}
+%endif
+
+%if %{with examples}
+%package examples
+Summary:        Data Plane Development Kit example applications
+BuildRequires:  libvirt-devel
+
+%description examples
+Example applications utilizing the Data Plane Development Kit, such
+as L2 and L3 forwarding.
+%endif
+
+%define sdkdir  %{_datadir}/%{name}
+%define docdir  %{_docdir}/%{name}
+%define incdir %{_includedir}/%{name}
+%define pmddir %{_libdir}/%{name}-pmds
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
```

Comment:
Can this be moved to the top section, where we usually have these tags?

Reply:
Removed duplicate.